Makefile

```diff
@@ -111,8 +111,17 @@ staticcheck:
 test:
 	$(GOTEST) -v ./...
 
+.PHONY: test-coverprofile
+test-coverprofile:
+	go test -covermode=count -coverprofile=coverage.out `go list ./... | grep -v integration` -json > test-report.out
+
+.PHONY: coverage
+coverage:
+	go test -covermode=count -coverprofile=coverage.out `go list ./... | grep -v integration`
+	go tool cover -html=coverage.out
+
 .PHOHY: checks
-checks: goimports-check vet generate tidy-check
+checks: goimports-check vet generate tidy-check test-coverprofile
```

Comment:
Not sure if it really fits into the `checks` target here. Should this be the one that `doesn't` generate the output file instead?
Do we need to add the one that does to the Github CI action?

Reply:
Both new actions generate an output file. 
test-coverprofile creates a json one which sonar cube can understtadnd
coverage creates one which the go cover tool can use

Reply:
We should also add the CI step, working out how to do that

Reply:
Ah, right. I'm still ot convinced it really fits into the `checks` target which so far only calls targets that perform static analysis and exit with a non-zero exit code on failure. The coverprofile would just create a file, but exit with 0 either way.

Reply:
Good point, I've removed it from that target.
Also added what I think is the required CI step